src/main/java/de/tum/in/www1/artemis/service/FileService.java

```diff
@@ -541,14 +546,15 @@ private boolean isIgnoredDirectory(final String filePath) {
      * @param targetDirectoryPath the path of the folder where the renamed folder should be located
      * @throws IOException if the directory could not be renamed.
      */
-    public void renameDirectory(String oldDirectoryPath, String targetDirectoryPath) throws IOException {
-        File oldDirectory = new File(oldDirectoryPath);
+
+    public void renameDirectory(Path oldDirectoryPath, Path targetDirectoryPath) throws IOException {
+        File oldDirectory = oldDirectoryPath.toFile();
```

Comment:
why do still use toFile here? We should use Files.exists instead in the if-condition